### PR TITLE
Fix zoom to layer/group not using proper index, not working on layer symbol legend items

### PR DESCRIPTION
**Pull request by nirvn**
@suricactus , follow up on your feature.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 59b00efe70aadb4dadec637eba2aa5af29b12374:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-18d9e897bc20cd57532d722b375b001fe2427fb9-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-18d9e897bc20cd57532d722b375b001fe2427fb9-arm-android.apk), [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-18d9e897bc20cd57532d722b375b001fe2427fb9-arm-neon-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-18d9e897bc20cd57532d722b375b001fe2427fb9-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-18d9e897bc20cd57532d722b375b001fe2427fb9-x86-android.apk)

**Comment by nirvn** [comment]
(backport not needed as the zoom to group functionality is on master only)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f00635f6281ca36c427ddb2c6fc646ad949fb243:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-a8b5a6e5d343c07296a0e869f2aa3c73181e0237-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-a8b5a6e5d343c07296a0e869f2aa3c73181e0237-arm-android.apk), [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-a8b5a6e5d343c07296a0e869f2aa3c73181e0237-arm-neon-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-a8b5a6e5d343c07296a0e869f2aa3c73181e0237-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-a8b5a6e5d343c07296a0e869f2aa3c73181e0237-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 30f8a40b12453ed9c93032b6bcd1ad2dfdfb796f:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-a2fe18997d74969b564ebbd4e5574624886c4d46-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-a2fe18997d74969b564ebbd4e5574624886c4d46-arm-android.apk), [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-a2fe18997d74969b564ebbd4e5574624886c4d46-arm-neon-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-a2fe18997d74969b564ebbd4e5574624886c4d46-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-a2fe18997d74969b564ebbd4e5574624886c4d46-x86-android.apk)

**Comment by suricactus** [comment]
Thanks for the fix of the index.

On reviving the legend "zoom to layer".

I am not super convinced about legend "zoom to layer", it is super confusing for me (and two other users I have seen). I never saw it as a feature, because there was no dedicated code for it, I thought it's more a strange side effect. I would expect to zoom only to features items with that legend symbol, but this expectations were wrong. 

On the other hand it looks strange without it. What if we change it "Zoom to containing/parent layer"?

So in general I'm -0.0 for it.

File: src/core/layertreemodel.cpp [hunk 1]
```diff
@@ -1091,9 +1099,18 @@ QgsRectangle FlatLayerTreeModelBase::nodeExtent( const QModelIndex &index, QgsQu
...
     QgsMapLayer *layer = QgsLayerTree::toLayer( node )->layer();
-
     if ( layer )
+    {
+      extent = mapSettings->mapSettings().layerToMapCoordinates( layer, layer->extent() );
+    }
+  }
+  else if ( QgsLayerTreeModelLegendNode *sym = mLayerTreeModel->index2legendNode( mapToSource( index ) ) )
+  {
+    QgsMapLayer *layer = qobject_cast<QgsMapLayer *>( sym->layerNode()->layer() );
```
Comment (m-kuhn) [inline src/core/layertreemodel.cpp:1109]: Isn:t layer() already returning a QgsMapLayer?

**Comment by nirvn** [comment]
+1 to rename button.

**Comment by nirvn** [comment]
@m-kuhn , needless casting removed.

@suricactus , button renamed to "zoom to parnet layer".

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a132ae07a12f94da01003a16b75f4bbeb9de8b6b:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-b16e679e82b257fb33d9763dcb5e454876665e85-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-b16e679e82b257fb33d9763dcb5e454876665e85-arm-android.apk), [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-b16e679e82b257fb33d9763dcb5e454876665e85-arm-neon-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-b16e679e82b257fb33d9763dcb5e454876665e85-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2910-b16e679e82b257fb33d9763dcb5e454876665e85-x86-android.apk)

**Review by suricactus — APPROVED** [review]
Good move.

**Pull request merged**